kedro/io/core.py

```diff
@@ -101,24 +101,36 @@ class AbstractDataSet(abc.ABC):
 
     Example:
     ::
-
-        >>> from kedro.io import AbstractDataSet
+        >>> from pathlib import Path, PurePosixPath
         >>> import pandas as pd
+        >>> from kedro.io import AbstractDataSet
+        >>>
         >>>
         >>> class MyOwnDataSet(AbstractDataSet):
-        >>>     def __init__(self, param1, param2):
+        >>>     def __init__(self, filepath, param1, param2=True):
+        >>>         self._filepath = PurePosixPath(filepath)
         >>>         self._param1 = param1
         >>>         self._param2 = param2
         >>>
         >>>     def _load(self) -> pd.DataFrame:
-        >>>         print("Dummy load: {}".format(self._param1))
-        >>>         return pd.DataFrame()
+        >>>         return pd.read_csv(self._filepath)
         >>>
         >>>     def _save(self, df: pd.DataFrame) -> None:
-        >>>         print("Dummy save: {}".format(self._param2))
+        >>>         df.to_csv(str(self._filepath))
+        >>>
+        >>>     def _exists(self) -> bool:
+        >>>         return Path(self._filepath).exists()
         >>>
         >>>     def _describe(self):
         >>>         return dict(param1=self._param1, param2=self._param2)
+
+    Example catalog.yml specification
+    ::
+        my_dataset:
```

Comment:
Does this actually render in the docs? I thought you'd need another blank line.

Reply:
Thanks @mzjp2!  It's been a long time since I have used restructured text.

Reply:
I think I got it fixed.  popped it into an online renderer and it seemed to like the latest change.